pkg/exchange/bybit/exchange.go

```diff
@@ -169,3 +175,82 @@ func (e *Exchange) QueryOpenOrders(ctx context.Context, symbol string) (orders [
 
 	return orders, nil
 }
+
+func (e *Exchange) SubmitOrder(ctx context.Context, order types.SubmitOrder) (*types.Order, error) {
+	if len(order.Market.Symbol) == 0 {
+		log.Warnf("empty symbol, orderClientId: %s", order.ClientOrderID)
+		return nil, fmt.Errorf("empty symbol")
+	}
+
+	req := e.client.NewPlaceOrderRequest()
+	req.Symbol(order.Symbol)
+
+	// set order type
+	orderType, err := toLocalOrderType(order.Type)
+	if err != nil {
+		return nil, err
+	}
+	req.OrderType(orderType)
+
+	// set side
+	side, err := toLocalSide(order.Side)
+	if err != nil {
+		log.Warnf("unexpected err: %v", err)
+		return nil, err
+	}
+	req.Side(side)
+
+	// set quantity
+	req.Qty(order.Market.FormatQuantity(order.Quantity))
+
+	// set price
+	switch order.Type {
+	case types.OrderTypeLimit:
+		req.Price(order.Market.FormatPrice(order.Price))
+	}
+
+	// set timeInForce
+	switch order.TimeInForce {
+	case types.TimeInForceFOK:
+		req.TimeInForce(bybitapi.TimeInForceFOK)
+	case types.TimeInForceIOC:
+		req.TimeInForce(bybitapi.TimeInForceIOC)
+	default:
+		req.TimeInForce(bybitapi.TimeInForceGTC)
+	}
+
+	// set client order id
+	if len(order.ClientOrderID) > maxOrderIdLen {
+		return nil, fmt.Errorf("unexpected length of order id, got: %d", len(order.ClientOrderID))
+	}
+	req.OrderLinkId(order.ClientOrderID)
+
+	if err := orderRateLimiter.Wait(ctx); err != nil {
+		log.WithError(err).Errorf("place order rate limiter wait error: %v", err)
+		return nil, err
+	}
+	res, err := req.Do(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(res.OrderId) == 0 || res.OrderLinkId != order.ClientOrderID {
+		log.Warnf("unexpected response, orderId: %s, orderLinkId: %s, clientOrderId: %s", res.OrderId, res.OrderLinkId, order.ClientOrderID)
+		return nil, fmt.Errorf("unexpected response")
```

Comment:
you can remove the warning and move the warning message to the error message, we can provide more details in the Errorf() call